The messages/ directory with its .mo/.po files, and the
top installation directory, do not require group write
permissions.

-- Matthias Andree (mandree@FreeBSD.org, 2020-04-14)

--- bin/check_perms.orig
+++ bin/check_perms
@@ -136,6 +136,14 @@
         elif (os.path.commonprefix((path, mm_cfg.QUEUE_DIR))
               == mm_cfg.QUEUE_DIR):
             targetperms = QFILEPERMS
+        elif (path == mm_cfg.PREFIX or os.path.split(path)[1] in
+              ('archives', 'bin', 'cgi-bin', 'cron', 'icons', 'mail', 'scripts',)
+              or '/messages' in path
+              or '/templates' in path
+              or '/tests' in path
+              or '/Mailman' in path):
+            # don't require LC_MESSAGES to be group writable
+            targetperms = DIRPERMS & ~S_IWGRP & ~S_ISGID
         else:
             targetperms = DIRPERMS
         octperms = oct(targetperms)
@@ -185,7 +193,7 @@
             if e.errno != errno.ENOENT: raise
             print(C_('WARNING: directory does not exist: %(d)s'))
             continue
-        if (mode & DIRPERMS) != DIRPERMS:
+        if (mode & DIRPERMS) != DIRPERMS and d != mm_cfg.PREFIX:
             STATE.ERRORS += 1
             print(C_('directory must be at least 02775: %(d)s'),)
             if STATE.FIX:
